stablehlo/dialect/TypeInference.cpp

```diff
@@ -3471,13 +3499,22 @@ LogicalResult verifyDynamicPadOp(std::optional<Location> location,
 
 LogicalResult verifyDynamicReshapeOp(std::optional<Location> location,
                                      Value outputShape, Value result) {
-  auto resultType = result.getType().dyn_cast<RankedTensorType>();
-  auto outputShapeType = outputShape.getType().dyn_cast<RankedTensorType>();
-  if (resultType && outputShapeType && outputShapeType.hasStaticShape() &&
+  auto resultType = result.getType().dyn_cast<ShapedType>();
+  auto outputShapeType = outputShape.getType().dyn_cast<ShapedType>();
+  if (resultType.hasRank() && outputShapeType.hasStaticShape() &&
```

Comment:
Deleted the `outputShapeType` and `resultType` null check? Should `dyn_cast` be `cast` if this is guaranteed?

Reply:
Yes, this is guaranteed. Good catch that this should be a `cast` - we actually have quite a few places in our codebase which are in a similar situation which I have all fixed now.